File: core/evidence/semantic_stability_limit_engine.py
from __future__ import annotations

from typing import Any, Dict


def semantic_stability_limits(stability: Dict[str, Any]) -> Dict[str, Any]:
    return {
        "max_confidence": stability.get("stability_limits", {}).get("max_confidence", 0.5),
        "expansion_allowed": stability.get("stable", False),
    }

File: src/evidence/semanticStabilityLimitEngine.ts
/**
 * Converted from Python: core/evidence/semantic_stability_limit_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function semanticStabilityLimits(stability: any): any {
  return {"max_confidence": py.get(py.get(stability, "stability_limits", {}), "max_confidence", py.F(0.5)), "expansion_allowed": py.get(stability, "stable", false)};
}
